pkg/kubernetes/status/health.go

```diff
@@ -21,19 +19,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
-	"github.com/kudobuilder/kudo/pkg/engine"
 	"github.com/kudobuilder/kudo/pkg/engine/resource"
 	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
 )
 
-func isJobTerminallyFailed(job *batchv1.Job) (bool, string) {
+func isJobTerminallyFailed(job *batchv1.Job) (bool, string, error) {
 	for _, c := range job.Status.Conditions {
 		if c.Type == batchv1.JobFailed && c.Status == corev1.ConditionTrue {
-			log.Printf("HealthUtil: Job %q has failed: %s", job.Name, c.Message)
-			return true, c.Message
+			return true, fmt.Sprintf("job %q has failed terminally: %s", job.Name, c.Message), nil
 		}
 	}
-	return false, ""
+	return false, "", nil
 }
 
 func IsDeleted(client client.Client, discovery discovery.CachedDiscoveryInterface, objs []runtime.Object) error {
```

Comment:
all these status APIs have changed to `bool, msg, error` except this one.   It is odd to have an `IsXYZ` without a bool... or no error means yes.

Reply:
To be honest, I wasn't fully sure if I should move this as well - It has a different signature, uses a client and fetches resources, works on a slice of objects, etc...

But I think it makes sense. I'll refactor it to work on a single object and adjust the signature to match the others.